assets/js/components/Dashboard/AlertDetails.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { ComponentType, useEffect, useState } from "react";
+import AlertCard from "./AlertCard";
+import { PlacesList } from "./PlacesPage";
+import { useNavigate, useParams } from "react-router-dom";
+import { Alert } from "../../models/alert";
+import { Button } from "react-bootstrap";
+import { ArrowLeft, ArrowUpRight } from "react-bootstrap-icons";
+import { formatEffect, placesWithSelectedAlert } from "../../util";
+import { useDashboardContext } from "./Dashboard";
+
+const AlertDetails: ComponentType = () => {
+  const { places, alerts, screensByAlertMap } = useDashboardContext();
+  const { id } = useParams();
+  const [selectedAlert, setSelectedAlert] = useState<Alert>();
+
+  const alertsUiUrl = document
+    .querySelector("meta[name=alerts-ui-url]")
+    ?.getAttribute("content");
+
+  const navigate = useNavigate();
```

Comment:
Does this just wrap `useNavigate()` in a variable? If so, what's the benefit we get from doing that?

Reply:
That is the only way to consume the hook from a function within a function component. Hooks can only be called from within custom hook or at the top level of a function component. So here, I save the hook logic in a variable for later and that allows me to call it from a function within the component.